src/main/kotlin/org/stellar/walletsdk/util/Sep24InteractiveFlow.kt

```diff
@@ -0,0 +1,127 @@
+ * @param server Horizon [Server] instance
+ * @param network Stellar [Network] instance
+ * @param httpClient HTTP client
+ *
+ * @return response object from the anchor
+ *
+ * @throws [AssetNotAcceptedForDepositException] if asset is not accepted for deposits
+ * @throws [AssetNotEnabledForDepositException] if asset is not enabled for deposits by the anchor
+ * @throws [AssetNotAcceptedForWithdrawalException] if asset is not accepted for withdrawals
+ * @throws [AssetNotEnabledForWithdrawalException] if asset is not enabled for withdrawals by the
+ * anchor
+ * @throws [NetworkRequestFailedException] if network request fails
+ */
+suspend fun interactiveFlow(
+  type: InteractiveFlowType,
+  accountAddress: String,
+  assetCode: String,
+  assetIssuer: String,
+  memoId: String? = null,
+  clientDomain: String? = null,
+  extraFields: Map<String, Any>? = null,
+  walletSigner: WalletSigner,
+  anchor: Anchor,
+  server: Server,
+  network: Network,
+  httpClient: OkHttpClient,
+): InteractiveFlowResponse {
+  return CoroutineScope(Dispatchers.IO)
+    .async {
+      val sep24RequiredFields =
+        listOf(
+          StellarTomlFields.SIGNING_KEY.text,
+          StellarTomlFields.TRANSFER_SERVER_SEP0024.text,
+          StellarTomlFields.WEB_AUTH_ENDPOINT.text
+        )
+      val toml = StellarToml(stellarAddress = assetIssuer, server, httpClient)
+      val tomlContent = toml.getToml()
+
+      // Check toml for required SEP-24 fields
+      toml.hasFields(fields = sep24RequiredFields, tomlContent = tomlContent)
+
+      val transferServerEndpoint =
+        tomlContent[StellarTomlFields.TRANSFER_SERVER_SEP0024.text].toString()
+
+      val serviceInfo = anchor.getServicesInfo(transferServerEndpoint)
+
+      // Check if deposit/withdraw is enabled for the asset
+      if (type == InteractiveFlowType.DEPOSIT) {
+        val asset =
+          serviceInfo.deposit[assetCode] ?: throw AssetNotAcceptedForDepositException(assetCode)
+
+        if (!asset.enabled) {
+          throw AssetNotEnabledForDepositException(assetCode)
+        }
+      } else {
+        val asset =
+          serviceInfo.withdraw[assetCode] ?: throw AssetNotAcceptedForWithdrawalException(assetCode)
+
+        if (!asset.enabled) {
+          throw AssetNotEnabledForWithdrawalException(assetCode)
+        }
+      }
+
+      val homeDomain = toml.getHomeDomain()
+
+      // SEP-10 auth
+      val auth =
+        Auth(
+          accountAddress = accountAddress,
+          webAuthEndpoint = tomlContent[StellarTomlFields.WEB_AUTH_ENDPOINT.text].toString(),
+          homeDomain = homeDomain,
+          clientDomain = clientDomain,
+          memoId = memoId,
+          networkPassPhrase = network.networkPassphrase,
+          walletSigner = walletSigner,
+          httpClient = httpClient
+        )
+      val authToken = auth.authenticate()
+
+      val requestParams = mutableMapOf<String, Any>()
+      requestParams["account"] = accountAddress
```

Comment:
We're assuming here that `accountAddress` is both the account that is authenticating as well as the account that will be used as the source of funds. SEP-24 allows these accounts to be different though.